lib/internal/modules/cjs/loader.js

```diff
@@ -840,20 +844,24 @@ const CircularRequirePrototypeWarningProxy = new Proxy({}, {
   }
 });
 
-// Object.prototype and ObjectProtoype refer to our 'primordials' versions
+// Object.prototype and ObjectPrototype refer to our 'primordials' versions
 // and are not identical to the versions on the global object.
 const PublicObjectPrototype = global.Object.prototype;
 
 function getExportsForCircularRequire(module) {
-  if (module.exports &&
-      ObjectGetPrototypeOf(module.exports) === PublicObjectPrototype &&
-      // Exclude transpiled ES6 modules / TypeScript code because those may
-      // employ unusual patterns for accessing 'module.exports'. That should be
-      // okay because ES6 modules have a different approach to circular
-      // dependencies anyway.
-      !module.exports.__esModule) {
-    // This is later unset once the module is done loading.
-    ObjectSetPrototypeOf(module.exports, CircularRequirePrototypeWarningProxy);
+  if (module.exports) {
+    const proxiedObject = getProxyDetails(module.exports, false) ||
+                          module.exports;
```

Comment:
We should benchmark this before landing just so we know what the impact is. Last I checked (and it was quite some time ago) `getProxyDetails` was rather expensive.

Reply:
Here is the benchmark: https://ci.nodejs.org/view/Node.js%20benchmark/job/benchmark-node-micro-benchmarks/594/console

It does not seem to slow down the code significantly. I am happy to run the benchmark again to get a higher confidence level in case it's requested.